src/RequestHeaderParser.php

```diff
@@ -25,9 +26,13 @@ public function feed($data)
         $this->buffer .= $data;
 
         if (false !== strpos($this->buffer, "\r\n\r\n")) {
-            list($request, $bodyBuffer) = $this->parseRequest($this->buffer);
+            try {
+                list($request, $bodyBuffer) = $this->parseRequest($this->buffer);
+                $this->emit('headers', array($request, $bodyBuffer));
+            } catch (Exception $exception) {
+                $this->emit('error', [$exception]);
```

Comment:
> Not sure this should be moved moved here, removing the listeners is not covered by any tests.

I'll update the tests tomorrow testing the removal of listeners.

> I agree that it makes sense to remove all listeners after an error, but we should probably look at line 20 then. Should this be postponed?

Makes more sense in a separate PR tbh.

> This whole block looks a bit unclear to me: catching exceptions from both parsing the request and emitting the `headers` event and continuing after an error only to remove all listeners.

It is controlling of the flow, if parsing the headers fails emitting will not be executed. To clear this up I'll move that to a new function instead.